Week4/ReadingList/app/src/main/java/com/florencenjeri/readinglist/BooksAdapter.kt

```diff
@@ -0,0 +1,63 @@
+package com.florencenjeri.readinglist
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+import com.florencenjeri.readinglist.model.Books
+
+class BooksAdapter(private val books: Array<Books>, private val clickListener: BooksListClickListener) :
+    RecyclerView.Adapter<BooksAdapter.BooksViewHolder>() {
+    interface BooksListClickListener {
+        fun listItemClicked(books: Books)
+
+    }
+
+    class BooksViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+
+        private var bookTitle: TextView = view.findViewById(R.id.book_title)
```

Comment:
Use Kotlin Android Extensions or ViewBindings to shorten this code and make it a bit nicer!